src/resolve/resolved-dns-server.c

```diff
@@ -909,7 +909,7 @@ DnsServer *manager_get_dns_server(Manager *m) {
                  * servers */
 
                 HASHMAP_FOREACH(l, m->links)
-                        if (l->dns_servers) {
+                        if (l->dns_servers && l->default_route) {
```

Comment:
So, when `l->default_route` is changed, `m->current_dns_server` should be unset, something like the way in `dns_server_new()`.